node/network/bridge/src/validator_discovery.rs

```diff
@@ -0,0 +1,588 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! A validator discovery service for the Network Bridge.
+
+use core::marker::PhantomData;
+use std::collections::{HashSet, HashMap, hash_map};
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use futures::channel::{mpsc, oneshot};
+
+use sc_network::Multiaddr;
+use sc_authority_discovery::Service as AuthorityDiscoveryService;
+use polkadot_node_network_protocol::PeerId;
+use polkadot_primitives::v1::{AuthorityDiscoveryId, Block, Hash};
+
+const PRIORITY_GROUP: &'static str = "parachain_validators";
```

Comment:
Should each protocol (collation, validation) have its own priority group?

Reply:
that's a good question, I'm not sure about this.
From the NetworkBridge perspective there is no difference. Or rather, what would be the benefit of separating the two?